### PR TITLE
fix: add testing code to the transformation action

### PR DESCRIPTION
**Pull request by abhimanyubabbar**
## Description of the change
Adding tests to the codebase

Fixes: DAT-608
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by abhimanyubabbar** [comment]
Screenshot 2024-09-16 at 3 03 31 PM

File: src/apiCalls.js [hunk 1]
```diff
@@ -10,35 +10,40 @@ const listTransformationsEndpoint = `${serverEndpoint}/transformations`;
...
 const defaultHeader = {
-  "user-agent": "transformationAction"
-}
+  "user-agent": "transformationAction",
+};
 
 async function getAllTransformations() {
+  core.info(`Getting all transformations from upstream`);
+
   return axios.default.get(listTransformationsEndpoint, {
```
Comment (Jayachand) [inline src/apiCalls.js:19]: minor: create a axios client (auth + headers) upfront and use it everywhere

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  let workspaceLibraries = [];
+
+  const transformationsResponse = await getAllTransformations();
+  workspaceTransformations = transformationsResponse.data
+    ? JSON.parse(JSON.stringify(transformationsResponse.data.transformations))
+    : [];
+
+  const librariesResponse = await getAllLibraries();
+  workspaceLibraries = librariesResponse.data
+    ? JSON.parse(JSON.stringify(librariesResponse.data.libraries))
```
Comment (Jayachand) [inline src/main.js:63]: parse and stringify looks redundant

Comment (abhimanyubabbar) [reply src/main.js:63]: This is done previously and didn't change anything regarding this.

File: src/apiCalls.js (hunk 1 above)
Comment (abhimanyubabbar) [reply src/apiCalls.js:19]: I can make this change.

**Review by Jayachand — APPROVED** [review]

**Pull request merged**